drivers/soundwire/intel.c

```diff
@@ -529,17 +529,24 @@ intel_pdi_alh_configure(struct sdw_intel *sdw, struct sdw_cdns_pdi *pdi)
 	intel_writel(alh, SDW_ALH_STRMZCFG(pdi->intel_alh_id), conf);
 }
 
-static int intel_config_stream(struct sdw_intel *sdw,
+static int intel_params_stream(struct sdw_intel *sdw,
```

Comment:
@plbossart when you say "Last, these structures are expanded to expose the link_id and alh_stream_id parameters which are needed to align with firmware IPC needs", its not clear which structures you meant here. You talk only about callbacks until this line and suddenly "these structures" seems a bit out of context.

Reply:
the previous paragraph states 'These two callbacks now rely on structures instead of lists of parameters.'

maybe 

These two callbacks now rely on structures which expose the link_id and alh_stream_id (required by the firmware IPC), instead of a list of parameters. This change was suggested on alsa-devel, along with the removal of the 'void *' definitions?

Would this help?

Reply:
@plbossart yes that definitely reads better.

Reply:
updated